### PR TITLE
Decompose snap_utility_allowance into three variables for SUA, LUA, and IUA

### LINKED ISSUE #3869: Decompose `snap_utility_allowance` into three variables for SUA, LUA, and IUA
And use `adds`. Code health only.

### PR DESCRIPTION
**Pull request by yy3187**
Fixes #3869

### CODE REVIEW

File: changelog_entry.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+- bump: minor
+  changes:
+    added:
+    - Louisiana adoption credit.
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:4]: adjust here

File: policyengine_us/data/datasets/cps/cps.py [hunk 2]
```diff
@@ -478,6 +478,58 @@ def add_previous_year_income(self, cps: h5py.File) -> None:
...
+        ~cps_cur_record_in_sample
+    ] = Y_pred.WSAL_VAL_prev.values
+    df["self_employment_income_prev"][
+        ~cps_cur_record_in_sample
+    ] = Y_pred.SEMP_VAL_prev.values
+
+    cps["employment_income_last_year"] = df["employment_income_prev"].values
+    cps["self_employment_income_last_year"] = df[
+        "self_employment_income_prev"
+    ].values
```
Comment (PavelMakarchuk) [inline policyengine_us/data/datasets/cps/cps.py:532]: delete this code

File: policyengine_us/tests/policy/baseline/gov/usda/snap/income/deductions/snap_individual_utility_allowance.yaml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+- name: Inidivudal Utility Allowance
+  period: 2023
+  input:
+    snap_utility_allowance_type: IUA
+    snap_utility_region_str: CA
+    phone_expense: 20
+  output:
+    snap_individual_utility_allowance: 228 # 19 * 12
```
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/usda/snap/income/deductions/snap_individual_utility_allowance.yaml:8]: add a couple of other tests, representing edge cases

File: policyengine_us/variables/gov/states/la/tax/income/la_refundable_credits.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/la/tax/income/la_refundable_credits.py:1]: did we intend to delete this?

File: policyengine_us/variables/gov/usda/snap/income/deductions/shelter/snap_individual_utility_allowance.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
+from policyengine_us.model_api import *
+
+
+class snap_individual_utility_allowance(Variable):
+    value_type = float
+    entity = SPMUnit
+    label = "Individual Utility Allowance"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/usda/snap/income/deductions/shelter/snap_individual_utility_allowance.py:7]:
```suggestion
    label = "SNAP Individual Utility Allowance"
```

Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/usda/snap/income/deductions/shelter/snap_individual_utility_allowance.py:7]: and others

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/gov/states/la/tax/income/la_refundable_credits.py
Comment (yy3187) [reply policyengine_us/variables/gov/states/la/tax/income/la_refundable_credits.py:1]: yes, it was the adoption credit.

Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/la/tax/income/la_refundable_credits.py:1]: I believe we need this back

File: changelog_entry.yaml
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]: test failing

File: policyengine_us/tests/policy/baseline/gov/states/la/tax/income/la_refundable_credits.py
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/la/tax/income/la_refundable_credits.py:1]: was this renamed?

Comment (PavelMakarchuk) [reply policyengine_us/tests/policy/baseline/gov/states/la/tax/income/la_refundable_credits.py:1]: This is causing the test issues

File: policyengine_us/variables/gov/usda/snap/income/deductions/shelter/snap_utility_allowance.py [hunk 5]
```diff
@@ -5,50 +5,11 @@ class snap_utility_allowance(Variable):
     value_type = float
     entity = SPMUnit
     label = "Standard Utility Allowance"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/usda/snap/income/deductions/shelter/snap_utility_allowance.py:7]:
```suggestion
    label = "Standard Utility Allowance"
    unit = USD
```

File: policyengine_us/variables/gov/usda/snap/income/deductions/shelter/snap_individual_utility_allowance.py [hunk 6]
```diff
@@ -0,0 +1,28 @@
+from policyengine_us.model_api import *
+
+
+class snap_individual_utility_allowance(Variable):
+    value_type = float
+    entity = SPMUnit
+    label = "SNAP Individual Utility Allowance"
+    unit = USD
+    documentation = "The regular utility allowance deduction for SNAP"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/usda/snap/income/deductions/shelter/snap_individual_utility_allowance.py:9]:
```suggestion
    documentation = "The individual utility allowance deduction for SNAP"
```

File: policyengine_us/variables/gov/usda/snap/income/deductions/shelter/snap_limited_utility_allowance.py [hunk 7]
```diff
@@ -0,0 +1,21 @@
+from policyengine_us.model_api import *
+
+
+class snap_limited_utility_allowance(Variable):
+    value_type = float
+    entity = SPMUnit
+    label = "SNAP Limited Utility Allowance"
+    unit = USD
+    documentation = "The regular utility allowance deduction for SNAP"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/usda/snap/income/deductions/shelter/snap_limited_utility_allowance.py:9]:
```suggestion
    documentation = "The limited utility allowance deduction for SNAP"
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/gov/states/la/tax/income/la_refundable_credits.py
Comment (MaxGhenis) [reply policyengine_us/variables/gov/states/la/tax/income/la_refundable_credits.py:1]: how does that connect to snap utility allowances?

**Review by MaxGhenis — APPROVED** [review]
